x-pack/plugins/infra/public/pages/metrics/hosts/components/tabs/tabs.tsx

```diff
@@ -5,37 +5,71 @@
  * 2.0.
  */
 
-import React from 'react';
-import { EuiTabbedContent, EuiSpacer, type EuiTabbedContentTab } from '@elastic/eui';
+import React, { useRef, useState } from 'react';
+import { EuiTabs, EuiTab, EuiSpacer } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
 import { MetricsGrid } from './metrics/metrics_grid';
+import { AlertsTabContent } from './alerts';
+
+import { AlertsTabBadge } from './alerts_tab_badge';
+import { tabIds } from './config';
+
+const tabs = [
+  {
+    id: tabIds.METRICS,
+    name: i18n.translate('xpack.infra.hostsViewPage.tabs.metricsCharts.title', {
+      defaultMessage: 'Metrics',
+    }),
+    'data-test-subj': 'hostsView-tabs-metrics',
+  },
+  {
+    id: tabIds.ALERTS,
+    name: i18n.translate('xpack.infra.hostsViewPage.tabs.alerts.title', {
+      defaultMessage: 'Alerts',
+    }),
+    append: <AlertsTabBadge />,
+    'data-test-subj': 'hostsView_tab_alerts',
+  },
+];
+
+const initialRenderedTabsSet = new Set([tabs[0].id]);
+
+export const Tabs = () => {
+  // This map allow to keep track of which tabs content have been rendered the first time.
+  // We need it in order to load a tab content only if it gets clicked, and then keep it in the DOM for performance improvement.
+  const renderedTabsSet = useRef(initialRenderedTabsSet);
+
+  const [selectedTabId, setSelectedTabId] = useState(tabs[0].id);
+
+  const tabEntries = tabs.map((tab, index) => (
+    <EuiTab
+      {...tab}
+      key={index}
+      onClick={() => {
+        renderedTabsSet.current.add(tab.id); // On a tab click, mark the tab content as allowed to be rendered
+        setSelectedTabId(tab.id);
+      }}
+      isSelected={tab.id === selectedTabId}
+      append={tab.append}
+    >
+      {tab.name}
+    </EuiTab>
+  ));
 
-interface WrapperProps {
-  children: React.ReactElement;
-}
-const Wrapper = ({ children }: WrapperProps) => {
   return (
     <>
-      <EuiSpacer size="s" />
-      {children}
+      <EuiTabs>{tabEntries}</EuiTabs>
+      <EuiSpacer />
+      {renderedTabsSet.current.has(tabIds.METRICS) && (
+        <div hidden={selectedTabId !== tabIds.METRICS}>
+          <MetricsGrid />
+        </div>
+      )}
+      {renderedTabsSet.current.has(tabIds.ALERTS) && (
+        <div hidden={selectedTabId !== tabIds.ALERTS}>
+          <AlertsTabContent />
+        </div>
+      )}
```

Comment:
Unmounting the component it's the natural behaviour for the `EuiTabbedContent` and I guess they meant to do it since it doesn't imply implicit memorization and, in case a dev need more control, can use the API we also switched to :)

Reply:
Yeah. I can see use-cases for both behaviours. It'd be nice if there was an option there to allow it to keep tabs mounted, though. But, for now, it doesn't seem like this has been an issue.